File: bigdata/scala/src/main/scala/cpt1/Student.scala
package cpt1

class Student(name: String, var age: Int){
  def print(): Unit={
    println( "name=" + name +", age=" + age + ", school=" + Student.school );
  }
}

//引入伴生对象
object Student{
  val school: String = "K"

  def main(args: Array[String]): Unit = {
    val a=new Student("a",20)
    val b = new Student("b", 14)
    a.print()
    b.print()
  }
}

File: scala/src/main/java/Student.java
//cpt1
public class Student {
    private String name;
    private Integer age;
    private static String school="K";

    public Student(String name, Integer age) {
        this.name = name;
        this.age = age;
    }

    public void print() {
        System.out.println( "Student{" +"name=" + this.name +", age=" + this.age + ", school=" + Student.school + '}');
    }

    public static void main(String[] args) {
        Student a=new Student("a",20);
        Student b = new Student("b", 14);
        a.print();
        b.print();
    }
}